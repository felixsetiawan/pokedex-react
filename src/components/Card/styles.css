.container {
    display: flex;
    border: solid 1px black;
}

img {
    max-width: 100%;
    height: auto;
}