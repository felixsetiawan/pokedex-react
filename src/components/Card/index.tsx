import React from "react";
import { IPokeDetailList } from "@/hooks/usePokemon";
import "./styles.css";

const Card = ({ data }: { data: IPokeDetailList }) => {
  return (
    <div className="container">
      <div>{data.id}</div>
      <div>{data.name}</div>
      <img
        src={data.image}
        width="200"
        height="200"
        alt={data.name}
        loading="lazy"
      />
    </div>
  );
};

export default Card;
