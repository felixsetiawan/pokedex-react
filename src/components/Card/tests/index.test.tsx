import React from "react";
import renderer from "react-test-renderer";
import Card from "../";

it("render Card correctly", () => {
  const component = renderer.create(
    <Card data={{ name: "name", id: 0, image: "" }} />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
