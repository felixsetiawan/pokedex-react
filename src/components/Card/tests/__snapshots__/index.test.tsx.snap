// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render Card correctly 1`] = `
<div
  className="container"
>
  <div>
    0
  </div>
  <div>
    name
  </div>
  <img
    alt="name"
    height="200"
    loading="lazy"
    src=""
    width="200"
  />
</div>
`;
